geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -114,19 +124,86 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     }, 0, 1, TimeUnit.SECONDS)
   }
 
-  override def run(): Unit = while (running.get) {
-    queue.take() match {
-      case update: CreateOrUpdate => featureCache.createOrUpdateFeature(update)
-      case del: Delete            => featureCache.removeFeature(del)
-      case clr: Clear             => featureCache.clear()
-      case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+  override def run(): Unit =
+    while (running.get) {
+      queue.take() match {
+        case update: CreateOrUpdate =>
+          fireEvent(KafkaFeatureEvent.changed(this, update.feature))
```

Comment:
The JavaDoc says

> Typically fired to signify that a change has occurred in the DataStore backing the FeatureSource. 

So I would think that the order should be reversed.  But, whatever you think...
If you wanted to reverse the order, just save the feature first (val f = ...; delete; fire(this, f))